nixos/modules/services/misc/gitea.nix

```diff
@@ -277,7 +278,35 @@ in
   };
 
   config = mkIf cfg.enable {
-    services.postgresql.enable = mkIf usePostgresql (mkDefault true);
+    services.postgresql = optionalAttrs (usePostgresql && cfg.database.createDatabase) {
+      enable = mkDefault true;
+
+      ensureDatabases = [ cfg.database.name ];
+      ensureUsers = [
+        { name = cfg.database.user;
+          ensurePermissions = { "DATABASE ${cfg.database.name}" = "ALL PRIVILEGES"; };
+        }
+      ];
+    };
+
+    services.mysql = optionalAttrs (useMysql && cfg.database.createDatabase) {
+      enable = mkDefault true;
+      package = mkDefault pkgs.mariadb;
+
+      ensureDatabases = [ cfg.database.name ];
+      ensureUsers = [
+        { name = cfg.database.user;
+          ensurePermissions = { "${cfg.database.name}.*" = "ALL PRIVILEGES"; };
+        }
+      ];
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.stateDir}' - ${cfg.user} gitea - -"
+      "d '${cfg.stateDir}/conf' - ${cfg.user} gitea - -"
```

Comment:
If you plan on using this to store config, it might be a good idea to either use `'${cfg.stateDir}/custom/conf'` or make sure Gitea takes the config file from `'${cfg.stateDir}/conf'` and does not look in `./custom/conf`. (That would need the cli flag `-c <path-to-config>` when running Gitea)

Reply:
@kolaente setting `environment.GITEA_WORK_DIR` in the systemd unit seems to make `gitea` automatically look for the `app.ini` file under `custom/conf/`.

Reply:
But then we don't need `${cfg.stateDir}/conf`? Or am I missing something here?

Reply:
> But then we don't need `${cfg.stateDir}/conf`? Or am I missing something here?

```
 	      # If we have a folder or symlink with gitea locales, remove it
		      # And symlink the current gitea locales in place
		      "L+ '${cfg.stateDir}/conf/locale' - - - - ${gitea.out}/locale"
```
So it appears that is where `gitea` looks for locales.

Reply:
Ah ok, didn't see that. Normally, it should not be necessary to do this, but since we're not building with `bindata`, that should be fine. Maybe to be changed in another pr.

Reply:
Sounds good. Thanks!